R/tabulate_survey.R

```diff
@@ -41,24 +45,33 @@ tabulate_survey <- function(x, var, strata = NULL, pretty = TRUE, digits = 1) {
     x <- srvyr::mutate(x, dummy = sprintf("%s %s", !! cod, !!st))
   }
   
-  p <- unique(dplyr::pull(srvyr::as_tibble(x), !!quote(dummy)))
-
   
 
   y <- srvyr::summarise(x, n = survey_total(var = "se", na.rm = TRUE))
+
+
+
+  # Here we pull out the relevant values for the proportions
+  p <- as.character(unique(dplyr::pull(srvyr::as_tibble(x), !!quote(dummy))))
+  p <- p[!is.na(p)]
   
-  res <- setNames(vector(mode = "list", length = nrow(x)), p)
+  # create an empty vector for the results
+  res <- setNames(vector(mode = "list", length = length(p)), p)
   
-  for (i in p) {
-    res[[p]] <- srvyr::summarise(x, 
-                                 proportion = srvyr::survey_mean(dummy == i, 
-                                                                 proportion = TRUE,
-                                                                 var = "ci",
-                                                                 na.rm = TRUE)
-                                 )
+  # loop over the values
+  for (i in p) { 
+    print(i)
+    if (y[y[[1]] == i, ]$n > 0) {
+      x <- srvyr::mutate(x, this = i)
+      res[[i]] <- survey::svyciprop(~I(dummy == this), x, method = "logit")
```

Comment:
not confident enough to start digging in all of this and change stuff ... but here, if you use srvyr::survey_mean and specify proportion= TRUE and prop_method, then it will use survey::svyciprop. 
Also the reason you got glm.fit error may be because of zero counts in a grop (log reg not converging) - maybe we need to consider one of the others. From reading on ze interwebz it seems like svyciprop has a method called "xlogit" which matches the CDC method of calculating CIs for surveys. This however wouldnt address our zeros issue... would probably be the beta or likelihood methods - to run a poisson or negative binomial type regression...

Reply:
I already [tried setting `proportion = TRUE`](https://github.com/R4EPI/sitrep/pull/98/commits/11638a0d528599818e122b7bd3b95ed9373aabaf#diff-5a50b278625ae42546a5708309ee2d8cR55), but it failed

The zero counts are exactly what's throwing everything off. As svyciprop states in the documenation: 

 > All methods undercover for probabilities close enough to zero or one, but "beta", "likelihood", "logit", and "logit" are noticeably better than the other two. None of the methods will work when the observed proportion is exactly 0 or 1.

I think for zero counts, we will simply have to report NA proportions.